File: arrays-hashing/medium/group-anagrams/sol.cs
// https://leetcode.com/problems/group-anagrams/

public class Solution
{
    public IList<IList<string>> GroupAnagrams(string[] strs)
    {
        var a = new Dictionary<string, List<string>>();

        foreach (string s in strs)
        {
            int[] freq = new int[26];
            foreach (char c in s)
            {
                freq[c - 'a']++;
            }
            string key = string.Join(',', freq);
            if (!a.ContainsKey(key))
            {
                a[key] = new List<string>();
            }
            a[key].Add(s);
        }

        return a.Values.ToList<IList<string>>();
    }
}

File: arrays-hashing/medium/group-anagrams/sol.js
// https://leetcode.com/problems/group-anagrams/

/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function (strs) {
  a = {};

  for (let s of strs) {
    freq = new Array(26).fill(0);
    for (let c of s) {
      freq[c.charCodeAt(0) - "a".charCodeAt(0)] += 1;
    }
    const key = freq.join(",");
    a[key] = a[key] ?? [];
    a[key].push(s);
  }

  return Object.values(a);
};
